utilities/tools/Set-Module.ps1

```diff
@@ -0,0 +1,144 @@
+#requires -version 7.3
+<#
+.SYNOPSIS
+Create/update all content of an AVM module that can be generated for the user
+
+.DESCRIPTION
+Create/update all content of an AVM module that can be generated for the user
+This includes
+- The `main.json` template(s)
+- The `README.md` file(s)
+
+.PARAMETER ModuleFolderPath
+Mandatory. The path to the module folder to generate the content for.
+
+.PARAMETER Recurse
+Optional. Set this parameter if you not only want to generate the content for one module, but also any nested module in the same path.
+
+.PARAMETER SkipBuild
+Optional. Set this parameter if you don't want to build/compile the JSON template(s) for the contained `main.bicep` file(s).
+
+.PARAMETER SkipReadMe
+Optional. Set this parameter if you don't want to generate the ReadMe file(s) for the module(s).
+
+.PARAMETER SkipFileAndFolderSetup
+Optional. Set this parameter if you don't want to setup the file & folder structure for the module(s).
+
+.PARAMETER ThrottleLimit
+Optional. The number of parallel threads to use for the generation. Defaults to 5.
+
+.PARAMETER ReadMeScriptFilePath
+Optional. The absolute path to the `Set-ModuleReadMe` script. Relevant only if `SkipReadMe` is not set and defaults to the default path of the script in the repository.
+
+.EXAMPLE
+Set-Module -ModuleFolderPath 'C:\avm\res\key-vault\vault'
+
+For the [key-vault\vault] module, build the Bicep module template & generate its ReadMe.
+
+.EXAMPLE
+Set-Module -ModuleFolderPath 'C:\avm\res\key-vault\vault' -Recurse
+
+For the [key-vault\vault] module or any of its children, build the Bicep module template & generate the ReadMe.
+
+.EXAMPLE
+Set-Module -ModuleFolderPath 'C:\avm\res\key-vault\vault' -Recurse -SkipReadMe
+
+For the [key-vault\vault] module or any of its children, build only the Bicep module template.
+
+.EXAMPLE
+Set-Module -ModuleFolderPath 'C:\avm\res' -Recurse
+
+For all modules in path [C:\avm\res], build the Bicep module template & generate the ReadMe.
+#>
+function Set-Module {
+
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $ModuleFolderPath,
+
+        [Parameter(Mandatory = $false)]
+        [switch] $Recurse,
+
+        [Parameter(Mandatory = $false)]
+        [switch] $SkipBuild,
+
+        [Parameter(Mandatory = $false)]
+        [switch] $SkipReadMe,
+
+        [Parameter(Mandatory = $false)]
+        [switch] $SkipFileAndFolderSetup,
+
+        [Parameter(Mandatory = $false)]
+        [int] $ThrottleLimit = 5,
+
+        [Parameter(Mandatory = $false)]
+        [string] $ReadMeScriptFilePath = (Join-Path (Get-Item $PSScriptRoot).Parent.FullName 'pipelines' 'sharedScripts' 'Set-ModuleReadMe.ps1')
+    )
+
+    # # Load helper scripts
```

Comment:
this is commented out. Is it needed?

Reply:
It will be. This code would be used to bootstrap a module. The problem: Our CI can currently not cope with that structure. We can remove the code too - but right now it's just here in alignment to the same tool `Set-AVMModule` in the Public Bicep Registry.